### PR TITLE
Feat: Added SignUp page

### PR DESCRIPTION
**Pull request by uday-nv-work**
Added Signup page with validations

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET = sjkelsadasdas;
```
Comment (DivyanshSareen) [inline .env]: please add .env file to .gitignore

File: src/components/Sidebar/Siderbar.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { SidebarData } from "./SidebarData";
+import profilePic from "assets/Profile Pic.png";
+import Logout from "@mui/icons-material/LogoutOutlined";
+import { useNavigate, useLocation } from "react-router-dom";
+import { useAuth } from "contexts/auth-context";
+export const Sidebar = () => {
+	const navigate = useNavigate();
+	const location = useLocation();
+	const { logoutHandler, user } = useAuth();
+	console.log(location);
```
Comment (DivyanshSareen) [inline src/components/Sidebar/Siderbar.jsx]: remove console.log() before merging to production

File: src/components/Sidebar/Siderbar.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	return (
+		<aside className='Home-left-container'>
+			<div className='Home-left-top'>
+				{SidebarData.map((val, key) => {
+					//Regex to match all the sub-routes in a Route.
+					let regex = new RegExp(val.link + "/?.*", "g");
+					console.log(regex, user);
```
Comment (DivyanshSareen) [inline src/components/Sidebar/Siderbar.jsx]: here as well

**Review by DivyanshSareen — COMMENTED** [review]
Loved the validate function and error handling. Good to merge!

**Review by srishtigautam21 — APPROVED** [review]
Good to merge

**Pull request merged**